### PR TITLE
migrate to actix-web 2.0

### PR DESCRIPTION
**Pull request by AnderEnder**

### CODE REVIEW

File: diamond-api/src/find.rs [hunk 1]
```diff
@@ -85,20 +85,29 @@ pub struct FindQuery {
...
-                Box::new(Json::<FindQuery>::from_request(req, payload).map(|x| x.into_inner()))
-            }
-            _ => Box::new(result(
-                Query::<FindQuery>::from_request(req, payload).map(|x| x.into_inner()),
-            )),
+            "application/x-www-form-urlencoded" => Form::<FindQuery>::from_request(req, payload)
+                .map(|r| match r {
+                    Ok(v) => Ok(v.into_inner()),
+                    Err(e) => Err(e.into()),
+                })
```
Comment (andy128k) [inline diamond-api/src/find.rs]: `|r| Ok(r?.into_inner())`

File: diamond-api/src/find.rs [hunk 2]
```diff
@@ -218,10 +227,7 @@ fn walk_tree(
     Ok(metrics)
 }
 
-pub fn find_handler(
-    ctx: Data<Context>,
-    query: FindQuery,
-) -> impl Future<Item = HttpResponse, Error = Error> {
+pub async fn find_handler(ctx: Data<Context>, query: FindQuery) -> Result<HttpResponse, Error> {
     match FindPath::from(&query) {
```
Comment (andy128k) [inline diamond-api/src/find.rs]: `let path = FindPath::from(&query).map_err(ErrorInternalServerError)?;`

File: diamond-api/src/render.rs [hunk 3]
```diff
@@ -176,10 +179,10 @@ pub struct RenderResponse {
     entries: Vec<Option<RenderResponseEntry>>,
 }
 
-pub fn render_handler(
+pub async fn render_handler(
     ctx: Data<Context>,
     query: RenderQuery,
-) -> impl Future<Item = HttpResponse, Error = failure::Error> {
+) -> Result<HttpResponse, failure::Error> {
     match Interval::new(query.from, query.until).map_err(err_msg) {
```
Comment (andy128k) [inline diamond-api/src/render.rs]: `match` -> `?`

File: diamond-api/src/bin/diamond-api.rs [hunk 4]
```diff
@@ -44,7 +44,7 @@ fn run(args: Args) -> io::Result<()> {
             .default_service(web::route().to(HttpResponse::NotFound))
     })
     .bind(listen)?
-    .run()?;
+    .run();
```
Comment (andy128k) [inline diamond-api/src/bin/diamond-api.rs]: Does it work? AFAIK `run()` just returns a future and quits.

File: diamond-api/src/find.rs (hunk 1 above)
Comment (andy128k) [reply diamond-api/src/find.rs]: This is just a suggestion. Feel free to ignoew

Comment (AnderEnder) [reply diamond-api/src/find.rs]: good catch.

File: diamond-api/src/find.rs (hunk 2 above)
Comment (AnderEnder) [reply diamond-api/src/find.rs]: This is an old code, but good catch

File: diamond-api/src/render.rs (hunk 3 above)
Comment (AnderEnder) [reply diamond-api/src/render.rs]: This is an old code, but good catch

File: diamond-api/src/bin/diamond-api.rs (hunk 4 above)
Comment (AnderEnder) [reply diamond-api/src/bin/diamond-api.rs]: 👍

Comment (AnderEnder) [reply diamond-api/src/bin/diamond-api.rs]: fixed with creating a system before it.

**Pull request merged**